TombEngine/Game/control/trigger.cpp

```diff
@@ -176,39 +177,8 @@ bool SwitchTrigger(short itemNumber, short timer)
 		return false;
 
 	//Handle Pulley
-	if (item.ObjectNumber == ID_PULLEY &&
-		item.ItemFlags[5] != 0 && item.ItemFlags[4] == 1 &&
-		player.Control.HandStatus != HandStatus::Busy)
-	{
-		item.Flags |= IFLAG_ACTIVATION_MASK;
-		item.Status = ITEM_ACTIVE;
-		item.ItemFlags[5] = 0;
-		return true;
-	}
-
-	if (item.ObjectNumber == ID_PULLEY &&
-		item.ItemFlags[5] != 0 && item.ItemFlags[4] == 0 &&
-		player.Control.HandStatus != HandStatus::Busy)
-	{
-		if (timer > 0)
-		{
-			item.Timer = timer;
-			item.Status = ITEM_ACTIVE;
-			item.ItemFlags[5] = 0;
-			if (timer != 1)
-				item.Timer = FPS * timer;
-
-			return true;
-		}
-
-		item.Flags |= IFLAG_ACTIVATION_MASK;
-		item.Status = ITEM_DEACTIVATED;
-		item.ItemFlags[5] = 0;
-		return true;
-	}
-
 	if (item.ObjectNumber == ID_PULLEY)
-		return false;
+		TriggerPulley(itemNumber, timer);
```

Comment:
You have moved pulley code to external function, but you ignore return value now.